### PR TITLE
Substitute trivial expressions

### PR DESCRIPTION
**Pull request by Enkelmann**
This PR adds a normalization pass to the `Project` struct that substitutes trivially computable expressions like `a XOR a` with their result.

The PR should only be merged after the **Fix implicit stores** PR is merged.

### CODE REVIEW

File: cwe_checker_rs/src/intermediate_representation/expression.rs [hunk 1]
```diff
@@ -76,6 +76,62 @@ impl Expression {
...
+        use Expression::*;
+        match self {
+            Var(_) | Const(_) | Unknown { .. } => (),
+            Subpiece {
+                low_byte,
+                size,
+                arg,
+            } => {
+                arg.substitute_trivial_operations();
+                if *low_byte == ByteSize::new(0) && *size == arg.bytesize() {
```
Comment (mellowCS) [inline cwe_checker_rs/src/intermediate_representation/expression.rs:92]: Just for my understanding. You'll simplify the subpiece operation when the size of the subpiece is 0? So you basically remove subpiece und just have the underlying expression?

File: cwe_checker_rs/src/pcode/expressions.rs [hunk 2]
```diff
@@ -122,6 +122,35 @@ pub struct Expression {
     pub input2: Option<Variable>,
 }
 
+impl From<Expression> for IrExpression {
+    /// Translates a P-Code expression into an expression of the internally used IR if possible.
+    /// Panics if translation is not possible,
+    /// e.g. for `LOAD`, `STORE` and and expressions that need the size of the output variable to be defined.
```
Comment (mellowCS) [inline cwe_checker_rs/src/pcode/expressions.rs:128]: I think there is an expression type missing

File: cwe_checker_rs/src/intermediate_representation/expression.rs (hunk 1 above)
Comment (Enkelmann) [reply cwe_checker_rs/src/intermediate_representation/expression.rs:92]: No, the size of the subpiece is the same as the size of the argument. And the subpiece starts at the zeroth byte of the argument (`low_byte = 0`). Or in other words, the subpiece is exactly the same as the argument.

File: cwe_checker_rs/src/pcode/expressions.rs (hunk 2 above)
Comment (Enkelmann) [reply cwe_checker_rs/src/pcode/expressions.rs:128]: You are right. I will clarify the doc comment!

File: test/src/lib.rs [hunk 3]
```diff
@@ -178,9 +178,8 @@ mod tests {
         mark_architecture_skipped(&mut tests, "mips64el"); // TODO: Check reason for failure!
         mark_architecture_skipped(&mut tests, "mips"); // TODO: Check reason for failure!
         mark_architecture_skipped(&mut tests, "mipsel"); // TODO: Check reason for failure!
-        mark_architecture_skipped(&mut tests, "ppc64"); // TODO: Check reason for failure!
-        mark_architecture_skipped(&mut tests, "ppc64le"); // TODO: Check reason for failure!
-        mark_architecture_skipped(&mut tests, "ppc"); // TODO: Check reason for failure!
+        mark_architecture_skipped(&mut tests, "ppc64"); // Ghidra generates mangled function names here for some reason.
+        mark_architecture_skipped(&mut tests, "ppc64le"); // Ghidra generates mangled function names here for some reason.
```
Comment (mellowCS) [inline test/src/lib.rs:182]: So we don't skip 'ppc' anymore?

Comment (Enkelmann) [reply test/src/lib.rs:182]: Yes. :-) The problem with 'ppc' was a trivial expression of the form `a OR a`, that the analysis could not handle. By substituting it away, the check now works for 'ppc'.

**Pull request merged**